src/controller/RentalsController.java

```diff
@@ -126,13 +132,18 @@ public static Rental getRental(int rid) {
 
         //TODO: This query is incorrect. It does not do what is expected
         String query = "SELECT * ";
-        if (location != null && city != null){
-            query = query + "FROM Rentals R, Vehicles V "+
-                    "WHERE R.fromDate =  " + date + " AND R.vlicense = V.vlicence AND V.city = " + city +
-                    " AND V.location = " + location;
+        if (location == null && city == null){
+            query = query +
+                    "FROM Rentals R, Vehicles V "+
+                    "WHERE R.fromDate =  '" + date + "' AND V.status = 'Rented' AND R.vlicense = V.vlicense AND V.city = '" + city +
```

Comment:
I think you should remove the V.status = 'Rented' because if you look for the daily rentals of a day in the past, a vehicle that is currently 'Available' wouldn't show up in those daily rentals even though it was rented at that point

Reply:
That makes sense.